bindings/native/go/lib.go

```diff
@@ -0,0 +1,164 @@
+package stronghold_go
+
+import (
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"github.com/awnumar/memguard"
+	"golang.org/x/crypto/blake2b"
+)
+
+type StrongholdNative struct {
+	ptr     StrongholdPointer
+	enclave *memguard.Enclave
+}
+
+func NewStronghold(key string) *StrongholdNative {
+	stronghold := &StrongholdNative{}
+	stronghold.enclave = memguard.NewEnclave([]byte(key))
+	return stronghold
+}
+
+func (s *StrongholdNative) Open(snapshotPath string) (bool, error) {
+	if s.ptr != nil {
+		return false, errors.New("snapshot is already open")
+	}
+
+	buffer, err := s.enclave.Open()
+	defer buffer.Destroy()
+
+	if err != nil {
+		return false, err
+	}
+
+	s.ptr = loadSnapshot(snapshotPath, buffer.String())
+
+	return true, nil
+}
+
+func (s *StrongholdNative) Create(snapshotPath string) (bool, error) {
+	if s.ptr != nil {
```

Comment:
I wonder, if it's possible to intercept functions calls always check for null pointers on default.

Reply:
I think that would be a good approach! `validate` would make the intend very clear :)